readthedocs/search/tests/test_faceted_search.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from readthedocs.search.documents import PageDocument
+
+
+class TestFileSearch(object):
+
+    @pytest.mark.parametrize('case', ['upper', 'lower', 'title'])
+    def test_search_exact_match(self, client, project, case):
+        """Check quoted query match exact phrase with case insensitively
+
+        Making a query with quoted text like ``"foo bar"`` should match
+        exactly ``foo bar`` or ``Foo Bar`` etc
+        """
+        # `Github` word is present both in `kuma` and `pipeline` files
+        # But the phrase Github can is available only in kuma docs.
+        # So search with this phrase to check
+        query_text = r'"GitHub can"'
+        cased_query = getattr(query_text, case)
+        query = cased_query()
+
+        page_search = PageDocument.faceted_search(query=query)
+        results = page_search.execute()
+
+        assert len(results) == 1
+        assert results[0]['project'] == 'kuma'
+        assert results[0]['path'] == 'documentation'
+
+    def test_search_combined_result(self, client, project):
+        """Check search result are combined of both `AND` and `OR` operator
+
+        If query is `Foo Bar` then the result should be as following order:
+
+        - Where both `Foo Bar` is present
+        - Where `Foo` or `Bar` is present
```

Comment:
Is the thought process that the first example -- both "foo" and "bar" are present -- will rank more highly than the second?

Reply:
Yes. It will have higher in rank as its more relevant